body/stretch_body/dynamixel_XL430.py

```diff
@@ -101,6 +104,20 @@
     4500000: 7
 }
 
+class DelayedKeyboardInterrupt:
+
+    def __enter__(self):
+        self.signal_received = False
+        self.old_handler = signal.signal(signal.SIGINT, self.handler)
+
+    def handler(self, sig, frame):
+        self.signal_received = (sig, frame)
+
+    def __exit__(self, type, value, traceback):
+        signal.signal(signal.SIGINT, self.old_handler)
+        if self.signal_received:
+            self.old_handler(*self.signal_received)
+
```

Comment:
Stretch ROS driver [errors out](https://gist.github.com/hello-fazil/cc6b69cba4ca5adb22b182cf361cadaf) when dynamixel groups are accessed with message `ValueError: signal only works in main thread`. This is because Python signals could only be used from the main thread, not external threads, i.e., simple action server. ([reference](https://stackoverflow.com/questions/54749342/valueerror-signal-only-works-in-main-thread-when-using-stomp)).

Reply:
Let's wrap the call to signal.signal in a try/catch where the ValueError is ignored. A KeyboardInterrupt wouldn't reach a python thread anyways, so there's no risk of the packet_handler becoming interrupted at the wrong time.

Reply:
Thanks for the suggestions. Appropriate exception handling is implemented and fixes the issues.